test/e2e_flags_test.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"flag"
+	"fmt"
+	"testing"
+)
+
+func assertAlsoAndV(t *testing.T, vwant string) {
+	alsologtostderr := flag.Lookup("alsologtostderr").Value.String()
+	if fmt.Sprintf("%t", true) != alsologtostderr {
+		t.Errorf("alsologtostderr was '%s', expected '%t'\n", alsologtostderr, true)
+	}
+	v := flag.Lookup("v").Value.String()
+	if vwant != v {
+		t.Errorf("alsologtostderr was '%s', expected '%s'\n", v, vwant)
+	}
+}
+
+func TestE2eFlags(t *testing.T) {
+	v := flag.Lookup("v")
+	if v == nil {
+		t.Fatal("Could not find 'v' flag; klog was not init-ed")
+	}
+	if v.Value.String() != "2" {
+		t.Fatalf("Either '%s' was passed in for v or v is not being initialized properly. The former is a test limitation, but the latter is an actual problem", v.Value.String())
+	}
+	if flagsSetup {
+		t.Error("SetupLoggingFlags should not have been called by the unit tests")
+	}
+
+	t.Log("Check default case with logverbose off")
```

Comment:
this can be handled by table drive test
https://github.com/golang/go/wiki/TableDrivenTests

Reply:
It takes an additional line of code to do that; still want me to?

diff --git a/test/e2e_flags_test.go b/test/e2e_flags_test.go
index 1fca733..c838511 100644
--- a/test/e2e_flags_test.go
+++ b/test/e2e_flags_test.go
@@ -45,25 +45,26 @@ func TestE2eFlags(t *testing.T) {
 		t.Error("SetupLoggingFlags should not have been called by the unit tests")
 	}
 
-	t.Log("Check default case with logverbose off")
-	flagsSetup = false
-	Flags.LogVerbose = false
-	flag.Set("alsologtostderr", "false")
-	SetupLoggingFlags()
-	assertAlsoAndV(t, "2")
-
-	t.Log("Check default case with logverbose on")
-	flagsSetup = false
-	Flags.LogVerbose = true
-	flag.Set("alsologtostderr", "false")
-	SetupLoggingFlags()
-	assertAlsoAndV(t, "8")
+	var flagTests = []struct {
+		log         string
+		logverbose  bool
+		nonDefaultV string
+		expectedV   string
+	}{
+		{"Check default case with logverbose off", false, "", "2"},
+		{"Check default case with logverbose on", true, "", "8"},
+		{"Check default case with logverbose on and v set to non-default value", true, "5", "5"},
+	}
 
-	t.Log("Check default case with logverbose on and v set to non-default value")
-	flagsSetup = false
-	Flags.LogVerbose = true
-	flag.Set("alsologtostderr", "false")
-	flag.Set("v", "5")
-	SetupLoggingFlags()
-	assertAlsoAndV(t, "5")
+	for _, tt := range flagTests {
+		t.Log(tt.log)
+		flagsSetup = false
+		Flags.LogVerbose = tt.logverbose
+		flag.Set("alsologtostderr", "false")
+		if tt.nonDefaultV != "" {
+			flag.Set("v", tt.nonDefaultV)
+		}
+		SetupLoggingFlags()
+		assertAlsoAndV(t, tt.expectedV)
+	}
 }

Reply:
It's much easier to read from my point of view. It also makes expanding test coverage easier to me. Btw this test only covers logverbose, any plan to cover all other flags defined here? I think one more testcase with all flags set would be good enough

Reply:
I don't think settings should be tested, just logic.